### PR TITLE
Make Endpoints broadcast as scalars

### PR DESCRIPTION
**Pull request by morris25**

### CODE REVIEW

File: test/endpoint.jl [hunk 1]
```diff
@@ -249,6 +249,9 @@ using Intervals: LeftEndpoint, RightEndpoint
 
         @test 1.0 != Endpoint(2, false)
         @test 1.0 != Endpoint(2, true)
+
+        result = 1.0 .== [Endpoint(1, false), Endpoint(1, true)]
+        @test result == [false, true]
```
Comment (omus) [inline test/endpoint.jl]: Should have a comment:

> Verify that `Endpoint` is treated as a scalar during broadcast

File: src/endpoint.jl [hunk 2]
```diff
@@ -34,6 +34,11 @@ function Base.hash(x::Endpoint{T, D}, h::UInt) where {T, D}
     return h
 end
 
+# https://github.com/JuliaLang/julia/pull/26601
+if VERSION >= v"0.7.0-DEV.4743"
```
[the remark this answers is not in the document]
Comment (omus) [reply src/endpoint.jl]: I don't think this change warrants dropping 0.6 support. In general we try to support the oldest version of Julia we can when it isn't problematic for open source. For our internal code there isn't a point to keeping 0.6 around anymore.

File: test/endpoint.jl (hunk 1 above)
Comment (omus) [reply test/endpoint.jl]: Without the comment I think I feel like I would refactor this code and possibly not test what was intended

File: src/endpoint.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (oxinabox) [reply src/endpoint.jl]: understood

**Pull request merged**